Std/Data/List/SplitOnList.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 Bulhwi Cha, Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bulhwi Cha, Mario Carneiro
+-/
+import Std.Data.Nat.Lemmas
+import Std.Data.List.Lemmas
+
```

Comment:
Could we have a module doc-string? I think it would be help to explain that the motivation is for specifying the behaviour of String.splitOn, even before that is actually implemented.

Reply:
I added it!